src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.LockTest.Unix.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Xunit;
+
+public class FileSystemWatcherLockTest
+{
+    [Fact]
+    public static void FileSystemWatcher_WatchLocksWatchedDirectory()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var fsw = new FileSystemWatcher())
+        {
+            fsw.Path = Path.GetFullPath(dir.Path);
+            fsw.EnableRaisingEvents = true;
+
+            Microsoft.Win32.SafeHandles.SafeFileHandle handle = Interop.Sys.Open(fsw.Path, Interop.Sys.OpenFlags.O_RDONLY, 0);
+            Assert.False(handle.IsInvalid); // We should always be able to open a handle
+            Assert.True(Interop.Sys.FLock(handle, Interop.Sys.LockOperations.LOCK_EX | Interop.Sys.LockOperations.LOCK_NB) < 0);
+
+            Interop.ErrorInfo err = Interop.Sys.GetLastErrorInfo();
+            Assert.True(err.Error == Interop.Error.EWOULDBLOCK); // validate it is a sharing violation
+        }
+    }
+}
```

Comment:
Ok, thanks for clarifying.

Given the underlying differences between the platforms, I don't think this adds a lot of value: I think the primary reason it exists on Windows is to prevent the directory from being deleted, and that simply doesn't apply here.

If you feel it adds enough value, though, I'm fine with you going ahead with it, with a shared rather than exclusive lock.  But I'd also be fine with you simply closing the PR and the issue.